flocker/ca/_ca.py

```diff
@@ -231,14 +267,90 @@ def write_credential_files(self, key_filename, certificate_filename):
             os.umask(original_umask)
 
 
-class NodeCredential(FlockerCredential):
+class UserCredential(PRecord):
+    """
+    A certificate for an API user, signed by a supplied certificate
+    authority.
+
+    :ivar FlockerCredential credential: The certificate and key pair
+        credential object.
+    :ivar bytes username: A username.
+    """
+    credential = field(mandatory=True, type=FlockerCredential)
+    username = field(mandatory=True, type=unicode)
+
+    @classmethod
+    def from_path(cls, path, username):
+        """
+        Load a node certificate from a specified path.
+
+        :param FilePath path: Directory where user certificate and key
+            files are stored.
+        :param bytes username: A UTF-8 encoded username.
+        """
+        username = unicode(username, "utf-8")
```

Comment:
Unless you have no idea what type an object is, `foo.decode("utf-8")` is a better way to make it unicode.